common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -7036,4 +7036,15 @@ public void syncFromConf(HiveConf conf) {
       set(e.getKey(), e.getValue());
     }
   }
+
+  public List<RegexMatchingConfEntry> getMatchingEntries(Pattern regex) {
+    List<RegexMatchingConfEntry> matchingEntries = new ArrayList<>();
+    for (Map.Entry<String, String> entry : this) {
+      Matcher matcher = regex.matcher(entry.getKey());
+      if (matcher.matches()) {
+        matchingEntries.add(new RegexMatchingConfEntry(entry.getKey(), matcher));
```

Comment:
why do we need this RegexMatchingConfEntry, can't we simply return the HashMap?

Reply:
Both the key and the created matcher are required in HiveServer2 class as well. I crated this wrapper In order to save resources by preventing the re-creation of the matcher there.

Reply:
i was only able to find `entry.getMatcher().group(0)`. Are we using `matches` on that matcher somewhere else?

Reply:
No